subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -105,6 +105,15 @@ void bl_boot(const struct fw_info *fw_info)
 	__set_MSP(vector_table[0]);
 	__set_PSP(0);
 
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SPLIM) && \
```

Comment:
No need for the HAS_SPLIM config, since this is a prerequisite for BUILTIN_STACK_GUARD in ARM.

Reply:
(for now)

Reply:
My thought was that if some other mechanism comes up in the future (other than SPLIM) then this would be future proof. IMO the code looks better if we ensure that SPLIM registers are present when doing direct access to them.

Reply:
OK, no strong opinion anyways. No harm if it stays.

Reply:
Reordered the checks